File: VB/App68/RichEditEx.vb
Imports Windows.Storage

Public Class RichEditEx
    Inherits RichEditBox

    Protected _Source As Uri

    Public Property Source() As Uri
        Get
            Return Me._Source
        End Get
        Set(ByVal value As Uri)
            Me._Source = value
            Me.OnSourceChanged()
        End Set
    End Property

    Private Async Sub OnSourceChanged()
        If Source Is Nothing Then
            Document.SetText(Windows.UI.Text.TextSetOptions.None, Nothing)
        Else
            Dim file As StorageFile = Await StorageFile.GetFileFromApplicationUriAsync(Source)
            Dim stream As Streams.IRandomAccessStream = Await file.OpenAsync(FileAccessMode.Read)
            Document.LoadFromStream(Windows.UI.Text.TextSetOptions.FormatRtf, stream)
        End If
    End Sub
End Class

File: CS/RichEditEx.cs
using System;
using Windows.Storage;
using Windows.UI.Xaml.Controls;

namespace App68 {
    public class RichEditEx : RichEditBox {

        protected Uri _Source;

        public Uri Source {
            get {
                return this._Source;
            }
            set {
                this._Source = value;
                this.OnSourceChanged();
            }
        }

        async void OnSourceChanged() {
            if (Source == null) {
                Document.SetText(Windows.UI.Text.TextSetOptions.None, null);
            }
            else {
                var file = await StorageFile.GetFileFromApplicationUriAsync(Source);
                var stream = await file.OpenAsync(Windows.Storage.FileAccessMode.Read);
                Document.LoadFromStream(Windows.UI.Text.TextSetOptions.FormatRtf, stream);
            }
        }
    }
}
